muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUserSchoolDataBridge.java

```diff
@@ -1095,30 +1157,47 @@ public List<UserPhoneNumber> listUserPhoneNumbers(SchoolDataIdentifier userIdent
     }
 
     PhoneNumber[] phoneNumbers = null;
-    Long pyramusStudentId = identifierMapper.getPyramusStudentId(userIdentifier.getIdentifier());
-    if (pyramusStudentId != null) {
-      phoneNumbers = pyramusClient.get(String.format("/students/students/%d/phoneNumbers", pyramusStudentId), PhoneNumber[].class);
-      if (phoneNumbers == null) {
-        return Collections.emptyList();
-      }
-    } else {
-      Long pyramusStaffId = identifierMapper.getPyramusStaffId(userIdentifier.getIdentifier());
-      if (pyramusStaffId != null) {
-        phoneNumbers = pyramusClient.get(String.format("/staff/members/%d/phoneNumbers", pyramusStaffId), PhoneNumber[].class);
-        if (phoneNumbers == null) {
-          return Collections.emptyList();
-        }
+    
+    PyramusUserType userType = identifierMapper.getIdentifierUserType(userIdentifier);
+
+    if (userType != null) {
+      switch (userType) {
+        case STUDENT:
+          Long pyramusStudentId = identifierMapper.getPyramusStudentId(userIdentifier.getIdentifier());
+          if (pyramusStudentId != null) {
+            phoneNumbers = pyramusClient.get(String.format("/students/students/%d/phoneNumbers", pyramusStudentId), PhoneNumber[].class);
+            if (phoneNumbers == null) {
+              return Collections.emptyList();
+            }
+          }
+        break;
+        
+        case STAFFMEMBER:
+          Long pyramusStaffId = identifierMapper.getPyramusStaffId(userIdentifier.getIdentifier());
+          if (pyramusStaffId != null) {
+            phoneNumbers = pyramusClient.get(String.format("/staff/members/%d/phoneNumbers", pyramusStaffId), PhoneNumber[].class);
+            if (phoneNumbers == null) {
+              return Collections.emptyList();
+            }
+          }
+        break;
+        
+        case STUDENTPARENT:
+          // Student parent has no phonenumbers at the moment.
```

Comment:
Couldn't this just `return Collections.emptyList();` then?

Reply:
Done